compiler/noirc_frontend/src/elaborator/expressions.rs

```diff
@@ -84,10 +84,10 @@ impl<'context> Elaborator<'context> {
                     expr_type: inner_expr_type.clone(),
                     expr_span: span,
                 });
+            }
 
-                if i + 1 == statements.len() {
-                    block_type = stmt_type;
-                }
+            if i + 1 == statements.len() {
+                block_type = stmt_type;
```

Comment:
This was a small unrelated error.
The if here shouldn't have been inside the other if, it was making blocks always return a Unit type.